crates/kubelet/src/grpc_sock/windows/mod.rs

```diff
@@ -6,41 +6,52 @@ use std::{
     task::{Context, Poll},
 };
 
-use futures::Stream;
+use futures::{FutureExt, Stream};
 use mio::Ready;
-use tokio::io::{AsyncRead, AsyncWrite};
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+use tokio_compat_02::FutureExt as CompatFutureExt;
 use tonic::transport::server::Connected;
 
-#[derive(Debug)]
 pub struct UnixStream {
-    inner: tokio::io::PollEvented<crate::mio_uds_windows::UnixStream>,
+    inner: tokio_compat_02::IoCompat<tokio_02::io::PollEvented<crate::mio_uds_windows::UnixStream>>,
 }
 
 impl UnixStream {
-    pub fn new(stream: crate::mio_uds_windows::UnixStream) -> Result<UnixStream, std::io::Error> {
-        Ok(UnixStream {
-            inner: tokio::io::PollEvented::new(stream)?,
-        })
+    pub async fn new(
+        stream: crate::mio_uds_windows::UnixStream,
+    ) -> Result<UnixStream, std::io::Error> {
+        let inner = async {
+            Ok::<_, std::io::Error>(tokio_compat_02::IoCompat::new(
+                tokio_02::io::PollEvented::new(stream)?,
+            ))
+        }
+        .compat()
+        .await?;
+        Ok(UnixStream { inner })
     }
 }
 
 pub struct Socket {
-    listener: tokio::io::PollEvented<crate::mio_uds_windows::UnixListener>,
+    listener: tokio_02::io::PollEvented<crate::mio_uds_windows::UnixListener>,
 }
 
 impl Socket {
-    pub fn new<P: AsRef<Path>>(path: &P) -> anyhow::Result<Self> {
-        let listener = crate::mio_uds_windows::UnixListener::bind(path)?;
-        Ok(Socket {
-            listener: tokio::io::PollEvented::new(listener)?,
-        })
+    #[allow(dead_code)]
```

Comment:
Probably a stupid question but why do we need to allow dead code here?

Reply:
Because of how it shows up as dead code when including this module in a non-testing scenario. We could put `cfg(test)` on this as an improvement